src/trigger.js

```diff
@@ -0,0 +1,189 @@
+try {
+  var m = require("mithril");
+} catch (e){
+}
+
+const app = {};
+
+app.view = function(vnode) {
+  const state = vnode.state;
+
+  const matchSearchKey = (project) => {
+    const keys = state.search_key.split(" ");
+    return keys.every((key) => {
+      const r = new RegExp(key, "i");
+      return project.path_with_namespace.match(r);
+    });
+  };
+
+  const projectEvents = (project) => {
+    const project_id = parseInt(project.id);
+    const config_project = state.config_projects[project_id] || {};
+    return config_project.events || {};
+  };
+  const branchList = () => {
+    if (!state.gitlab.branchs) {
+      return m("li", m("div","------"));
+    }
+    return state.gitlab.branchs.filter((branch) =>{ 
+      return branch;
+    }).map((branch) => {
+      return m("li",  m("a", {
+        onclick: (e) => {
+          var targ = e.target;
+          $( "#curBranch" ).html( targ.innerText );
```

Comment:
Please don't access Real DOM from Virtual DOM
(Don't use `$` method of jQuery )